parse_tz.c

```diff
@@ -159,6 +159,9 @@ static int read_preamble(const unsigned char **tzf, timelib_tzinfo *tz, unsigned
 		*type = TIMELIB_TZINFO_ZONEINFO;
 		return read_tzif_preamble(tzf, tz);
 	} else {
+		// This is the default value used to avoid this compilation error with `-Os` flag:
+		// "'type' may be used uninitialized in this function [-Werror=maybe-uninitialized]"
+		*type = TIMELIB_TZINFO_ZONEINFO;
```

Comment:
Wouldn't it be nicer if timelib just would set the default value in `timelib_parse_tzfile` on line 657?

```
656     int transitions_result, types_result;                                                                                           
657     unsigned int type; /* TIMELIB_TZINFO_PHP or TIMELIB_TZINFO_ZONEINFO */                                                          
658
```

Reply:
Thanks @derickr. Yeah, that also solves the issue. Pushed a fix.

BTW, do you think it's feasible to get a new tag after this gets merged with master? The embedded MongoDB builds fail without this.